pkg/cel/cel.go

```diff
@@ -0,0 +1,141 @@
+package cel
+
+import (
+	"fmt"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	"github.com/google/cel-go/interpreter/functions"
+	exprpb "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+
+	"github.com/blang/semver/v4"
+)
+
+// PropertiesKey is the key for bundle properties map (input data for CEL evaluation)
+const PropertiesKey = "properties"
+
+type Constraint struct {
+	// Constraint message that surfaces in resolution
+	// This field is optional
+	Message string `json:"message" yaml:"message"`
+
+	// The cel struct that contraints CEL expression
+	// This field is required
+	Cel *Cel `json:"cel" yaml:"cel"`
+}
+
+type Cel struct {
+	// The CEL expression
+	Rule string `json:"rule" yaml:"rule"`
+}
+
+type Evaluator interface {
+	Evaluate(env map[string]interface{}) (bool, error)
+}
+
+type EvaluatorProvider interface {
+	Evaluator(rule string) (Evaluator, error)
+}
```

Comment:
It would be good to get some GoDoc on these types to make it more obvious what their purpose is.

Are these specific to `cel`? If not, should they be in another package? Or should we name this package something else (e.g. `constraint`)?

Reply:
On the OLM PR, I call this package `constraints` but in api repo, I'm not sure if I should be more explicit about the constraint type. So I call it `cel` here but if `constraints` sounds more appropriate then I'm totally fine wit that.

Reply:
Yep, if this package is renamed to `constraint` or `constraints`, it would make sense to me to have something like:
```
$ tree constraint
constraint
├── cel.go (cel-specific stuff here)
├── compound.go (All, Any, None constraints here)
└── constraint.go (common types and interfaces, e.g. Constraint and Evaluator*)
```

WDYT?

Reply:
Actually, looking more closely at the evaluator interface, it looks cel-specific (at least for now)?

I'm curious if there's a more generic constraint interface that all of the constraint types could implement? When @estroz goes to add the compound constraint struct, will there be an interface he can implement such that he doesn't have to change anything on the OLM side?

Reply:
The original code was meant to use with `Evaluator()` function below as a "selector" on which provider to use via `id` parameter (if you remember the original EP with `evaluator` struct with `id` field). Now that we don't have `id` part anyway, this provider becomes CEL-specific which is fine. If we introduce a new "provider" later, we can simply create a new `NewSomethingEvaluatorProvider` and add it to the `SatResolver`. Now `evaluatorProvider` on SatResolver is a field with a single provider but it can be a collection of providers. That shouldn't be a problem code-wise later on.